### PR TITLE
Raise an error if someone tries to use the log_file config option

### PR DESCRIPTION
**Pull request by richvdh**
This has caused some confusion for people who didn't notice it going away.

cf #5678

### CODE REVIEW

File: synapse/config/logger.py [hunk 1]
```diff
@@ -81,11 +81,18 @@
 """
 )
 
+NO_LOG_FILE_ERROR = """\
+Support for the log_file configuration option and --log-file commandline option was
```
Comment (babolivier) [inline synapse/config/logger.py]:
```suggestion
Support for the log_file configuration option and --log-file command line option was
```

File: synapse/config/logger.py [hunk 2]
```diff
@@ -81,11 +81,18 @@
 """
 )
 
+NO_LOG_FILE_ERROR = """\
```
Comment (babolivier) [inline synapse/config/logger.py]: I find this name confusing, I would understand it as an error you raise when there's no `log_file`/`--log-file` provided (but it's actually the opposite). Maybe rename it in `LOG_FILE_ERROR`?

File: synapse/config/logger.py (hunk 1 above)
Comment (richvdh) [reply synapse/config/logger.py]: hrrmm maybe. I'll compromise on `command-line` :)

File: synapse/config/logger.py (hunk 2 above)
Comment (richvdh) [reply synapse/config/logger.py]: yup, fair. fixed.

File: synapse/config/logger.py (hunk 1 above)
Comment (babolivier) [reply synapse/config/logger.py]: Works for me :p

Comment (babolivier) [reply synapse/config/logger.py]: Works for me :p

**Pull request merged**